charts/posthog/templates/_postgresql.tpl

```diff
@@ -99,29 +100,50 @@ Set postgres port
 */}}
 {{- define "posthog.postgresql.port" -}}
 {{- if .Values.postgresql.enabled -}}
-    5432
+5432
 {{- else -}}
-{{- default "5432" .Values.postgresql.postgresqlPort -}}
+{{- .Values.externalPostgresql.postgresqlPort -}}
 {{- end -}}
 {{- end -}}
 
 {{/*
-Set postgres password
+Set postgres username
 */}}
-{{- define "posthog.postgresql.password" -}}
-{{ .Values.postgresql.postgresqlPassword | default "" }}
+{{- define "posthog.postgresql.username" -}}
+{{- if .Values.postgresql.enabled -}}
+"postgres"
+{{- else -}}
+{{- .Values.externalPostgresql.postgresqlUsername | quote -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Set postgres database
+*/}}
+{{- define "posthog.postgresql.database" -}}
+{{- if .Values.postgresql.enabled -}}
+{{- .Values.postgresql.postgresqlDatabase | quote -}}
+{{- else -}}
+{{- .Values.externalPostgresql.postgresqlDatabase | quote -}}
+{{- end -}}
 {{- end -}}
 
 {{/*
-Set postgres password b64
+Set postgres password. Don't use this outside of secrets!
 */}}
-{{- define "posthog.postgresql.passwordb64" -}}
-{{ .Values.postgresql.postgresqlPassword | default "" | b64enc | quote }}
+{{- define "posthog.postgresql.password" -}}
+{{- if and .Values.postgresql.enabled (not .Values.postgresql.existingSecret) -}}
+{{- .Values.postgresql.postgresqlPassword -}}
+{{- else if and (not .Values.postgresql.enabled) (not .Values.externalPostgresql.existingSecret) -}}
+{{- .Values.externalPostgresql.postgresqlPassword -}}
+{{- end -}}
 {{- end -}}
 
 {{/*
-Set postgres URL
+Set postgres secret created in secrets.yaml
```

Comment:
I think we should not set this anymore and simply provide the password via env variables (like we do for redis)

Reply:
`charts/posthog/templates/_snippet-redis-env.tpl` does exactly the opposite - it defines the password via secrets

Reply:
From our call: The comment was about using a different secrets file, rather than main secrets.yaml